cfavml-complex/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+#[cfg(target_arch = "x86")]
+use core::arch::x86::*;
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64::*;
+
+use cfavml::danger::{Avx2, SimdRegister};
+use num_complex::Complex;
+pub trait ComplexOps<T>
+where
+    Complex<T>: Copy,
+{
+    type Register: Copy;
+    /// Extracts the real component from a register
+    unsafe fn real_values(value: Self::Register) -> Self::Register;
+    /// Extracts the imaginary component from a register
+    unsafe fn imag_values(value: Self::Register) -> Self::Register;
+
+    /// Duplicates the real and imaginary components of a the input into separate registers
+    unsafe fn duplicate_complex_components(
+        value: Self::Register,
+    ) -> (Self::Register, Self::Register) {
+        (Self::real_values(value), Self::imag_values(value))
+    }
+    /// Swaps the real and imaginary components of a complex number
+    unsafe fn swap_complex_components(value: Self::Register) -> Self::Register;
+}
+
+pub struct Avx2Complex;
```

Comment:
I guess orphan rules are causing issues using the main exported types :/ Maybe the wrapper impl should be like `Complex<RegisterType>` so `Complex<Avx2>` etc...